pkg/ui/react-app/src/thanos/pages/blocks/BlockDetails.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { FC } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTimes } from '@fortawesome/free-solid-svg-icons';
+import { Block } from './block';
+import styles from './blocks.module.css';
+import moment from 'moment';
+
+interface BlockDetailsProps {
+  block: Block | undefined;
+  selectBlock: React.Dispatch<React.SetStateAction<Block | undefined>>;
+}
+
+export const BlockDetails: FC<BlockDetailsProps> = ({ block, selectBlock }) => {
+  return (
+    <div className={`${styles.blockDetails} ${block && styles.open}`}>
+      {block && (
+        <>
+          <div className={styles.detailsTop}>
+            <p>{block.ulid}</p>
+            <button className={styles.closeBtn} onClick={(): void => selectBlock(undefined)}>
```

Comment:
The close button is a bit too big, at least on my screen. Sometimes it doesn't even show up because of how big it is. Also, maybe it would be nice to make BlockDetails close automatically whenever the user clicks somewhere on the page?

Reply:
Oh, I'd lower the size of the icon, maybe we can use `&times;` native HTML entity instead of the icon.

> Also, maybe it would be nice to make BlockDetails close automatically whenever the user clicks somewhere on the page?

I think it wouldn't be that straightforward to implement as we'd need to use a click handler on everything except blocks. I'd look into it further.